desktop/core/src/desktop/lib/importer/api.py

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/env python
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from functools import wraps
+
+from rest_framework import status
+from rest_framework.decorators import api_view, parser_classes
+from rest_framework.parsers import JSONParser, MultiPartParser
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from desktop.lib.importer import operations
+from desktop.lib.importer.serializers import (
+  GuessFileHeaderSerializer,
+  GuessFileMetadataSerializer,
+  LocalFileUploadSerializer,
+  PreviewFileSerializer,
+  SqlTypeMapperSerializer,
+)
+
+LOG = logging.getLogger()
+
+
+# TODO: Improve error response further with better context -- Error UX Phase 2
+def api_error_handler(fn):
+  """
+  Decorator to handle exceptions and return a JSON response with an error message.
+  """
+
+  @wraps(fn)
+  def decorator(*args, **kwargs):
+    try:
+      return fn(*args, **kwargs)
+    except Exception as e:
+      LOG.exception(f"Error running {fn.__name__}: {str(e)}")
+      return Response({"error": str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+  return decorator
+
+
+@api_view(["POST"])
+@parser_classes([MultiPartParser])
+@api_error_handler
+def local_file_upload(request: Request) -> Response:
+  """Handle the local file upload operation.
+
+  This endpoint allows users to upload a file from their local system.
+  Uploaded file is validated using LocalFileUploadSerializer and processed using local_file_upload operation.
+
+  Args:
+    request: Request object containing the file to upload
+
+  Returns:
+    Response containing the result of the local upload operation, including:
+      - file_path: Path where the file was saved (if successful)
+
+  Note:
+    - File size limits apply based on MAX_LOCAL_FILE_SIZE_UPLOAD_LIMIT configuration.
+    - File type restrictions apply based on RESTRICT_LOCAL_FILE_EXTENSIONS configuration.
+  """
+
+  serializer = LocalFileUploadSerializer(data=request.data)
+
+  if not serializer.is_valid():
+    return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Add log statement to write error to rungunicornserver without asking end user to get HAR file from browser.

Reply:
These will be logged IIUC. Will recheck the package methods again.